packages/core/src/com/types.ts

```diff
@@ -51,10 +51,13 @@ export type UnknownFunction = (...args: unknown[]) => unknown;
 export type AsyncApi<T> = {
     [P in keyof T]: P extends keyof ServiceConfig<T>
         ? MultiTanentProxyFunction<T, P extends string ? P : never>
+        : T[P] extends (...args: any[]) => PromiseLike<any>
+        ? T[P]
         : T[P] extends (...args: infer Args) => infer R
-        ? (...args: Args) => ValuePromise<R>
+        ? (...args: Args) => Promise<R>
```

Comment:
isn't this the only place that uses ValuePromise?

Reply:
also used in `multiTenantMethod` in `service-config.ts`. might be a good idea to remove it in the future though.